pkg/controller/replication_controller.go

```diff
@@ -204,6 +204,10 @@ func (rm *ReplicationManager) getPodControllers(pod *api.Pod) *api.ReplicationCo
 // When a pod is created, enqueue the controller that manages it and update it's expectations.
 func (rm *ReplicationManager) addPod(obj interface{}) {
 	pod := obj.(*api.Pod)
+	if pod.DeletionTimestamp != nil {
```

Comment:
Comments updated.  I also added stronger checking and reset behavior to ValidateObjectMetaUpdate (deletionGracePeriodSeconds is immutable) and rest.BeforeCreate() to guard against objects being created with those values.

----- Original Message -----
> > @@ -204,6 +204,10 @@ func (rm *ReplicationManager) getPodControllers(pod
> > *api.Pod) *api.ReplicationCo
> >  // When a pod is created, enqueue the controller that manages it and
> >  update it's expectations.
> >  func (rm *ReplicationManager) addPod(obj interface{}) {
> >  	pod := obj.(*api.Pod)
> > +	if pod.DeletionTimestamp != nil {
> 
> ah, then we'll never get add pods with the deletion timestamp set unless you
> restart the controller manager or something right? (we get add pod events
> when the pod isn't in the store but is present in whatever the apiserver
> sends us -- this could either be a watch event or the output of a relist
> --). Please add that as a comment here.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/8641/files#r31480606